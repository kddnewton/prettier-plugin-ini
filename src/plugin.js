import languages from "./languages.js";
import parser from "./parser.js";
import printer from "./printer.js";

const plugin = {
  languages,
  parsers: {
    ini: parser
  },
  printers: {
    ini: printer
  },
  options: {
    iniSpaceAroundEquals: {
      type: "boolean",
      category: "INI",
      default: false,
      description:
        "Adds a space around the equals sign when specifying params.",
      since: "0.3.0"
    }
  },
  defaultOptions: {
    printWidth: 80,
    tabWidth: 2
  }
};

export default plugin;
