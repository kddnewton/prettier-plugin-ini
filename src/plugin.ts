import type { Plugin } from "prettier";

import parser from "./parser";
import printer from "./printer";

const plugin: Plugin = {
  languages: [
    {
      name: "INI",
      parsers: ["ini"],
      extensions: [".ini"],
      vscodeLanguageIds: ["ini"]
    }
  ],
  parsers: {
    ini: parser
  },
  printers: {
    ini: printer
  },
  options: {
    iniSpaceAroundEquals: {
      type: "boolean",
      category: "INI",
      default: false,
      description:
        "Adds a space around the equals sign when specifying params.",
      since: "0.3.0"
    }
  },
  defaultOptions: {
    printWidth: 80,
    tabWidth: 2
  }
};

export default plugin;
