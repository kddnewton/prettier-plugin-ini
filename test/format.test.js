import { readFileSync } from "fs";
import prettier from "prettier";
import plugin from "../src/plugin.js";

const fixtureFiles = ["with-sections", "no-sections"];

function format(content, opts) {
  return prettier.format(content, {
    parser: "ini",
    plugins: [plugin],
    ...opts
  });
}

fixtureFiles.forEach((fixtureFile) => {
  const fixturePath = new URL(`./fixtures/${fixtureFile}.ini`, import.meta.url);
  const fixture = readFileSync(fixturePath, "utf-8");

  describe(`${fixtureFile}:`, () => {
    test("defaults", async () => {
      const formatted = await format(fixture, { iniSpaceAroundEquals: false });
      expect(formatted).toMatchSnapshot();
    });

    test("with space around equals", async () => {
      const formatted = await format(fixture, { iniSpaceAroundEquals: true });
      expect(formatted).toMatchSnapshot();
    });
  });
});
