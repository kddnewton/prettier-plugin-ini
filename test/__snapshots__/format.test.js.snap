// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no-sections: defaults 1`] = `
"; last modified 1 April 2001 by John Doe
name=John Doe
organization=Acme Widgets Inc.
; use IP address in case network name resolution is not working
server=192.0.2.62
port=143
file=\\"payroll.dat\\"
# this is another kind of comment
"
`;

exports[`no-sections: with space around equals 1`] = `
"; last modified 1 April 2001 by John Doe
name = John Doe
organization = Acme Widgets Inc.
; use IP address in case network name resolution is not working
server = 192.0.2.62
port = 143
file = \\"payroll.dat\\"
# this is another kind of comment
"
`;

exports[`with-sections: defaults 1`] = `
"; last modified 1 April 2001 by John Doe

[owner]
name=John Doe
organization=Acme Widgets Inc.

[database]
; use IP address in case network name resolution is not working
server=192.0.2.62
port=143
file=\\"payroll.dat\\"
# this is another kind of comment
"
`;

exports[`with-sections: with space around equals 1`] = `
"; last modified 1 April 2001 by John Doe

[owner]
name = John Doe
organization = Acme Widgets Inc.

[database]
; use IP address in case network name resolution is not working
server = 192.0.2.62
port = 143
file = \\"payroll.dat\\"
# this is another kind of comment
"
`;
